File: yelingzi-blog/src/main/java/com/yeling/article/domian/entity/Article.java
package com.yeling.article.domian.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Article {

    private Integer id;

    private String nickname;//发布者名称

    private String title;//文章标题

    private String content;//文章内容

    private String articleCover;//文章封面

    private String category;//文章分类

    private String articleUrl;//文章地址

    private Integer starCount;

    private Integer state;

    private LocalDateTime createTime; //创建时间

    private LocalDateTime updateTime;//更新时间

    private Integer readCount; //阅读量

    private Integer commentCount;

    private Integer userId;

    private String userAvatar;

    private String tagList;

    private Integer likeCount;

    private Integer isOriginal;

    private String originalUrl;

    private Integer isTop;

    private String brief;

}

File: blog/src/types/article.ts
export interface ArticleList {
  /**
   * 文章id
   */
  id: number
  /**
   * 文章缩略图
   */
  articleCover: string
  /**
   * 文章标题
   */
  title: string
  /**
   * 文章摘要
   */
  articleDesc: string
  /**
   * 文章分类
   */
  articleClassify: Tag
  /**
   * 文章标签
   */
  tagList: Tag[]
  /**
   * 是否置顶 (0否 1是)
   */
  isTop: number
  /**
   * 发表时间
   */
  createTime: string
  /**
   * 更新时间
   */
  updateTime: string
}

export interface Article {
  id: number;
  nickname: string;
  title: string;
  content: string;
  articleCover: string;
  category: Category;
  starCount: number;
  state: number;
  createTime: string;
  updateTime: string;
  readCount: number;
  commentCount: number;
  userId: number;
  userAvatar: string;
  tagList: Tag[];
  likeCount: number;
  isOriginal: number;
  originalUrl: string;
  isTop: number;
  isLike: boolean
  isStar: boolean
}

export interface ArticleRecommend {
  id: number
  articleCover: string
  title: string
  createTime: string
}
export interface Tag {
  /**
   * 标签id
   */
  id: number
  /**
   * 标签名
   */
  tagName: string
}

export interface Category {
  id: number;
  categoryName: string;
}

export interface Archives {
  id: number
  title: string
  createTime: string
}
